### PR TITLE
Httpcore 618 http status as a class

### PR DESCRIPTION
**Pull request by garydgregory**
https://issues.apache.org/jira/browse/HTTPCORE-618

### CODE REVIEW

File: httpcore5/src/main/java/org/apache/hc/core5/http/HttpStatus.java [hunk 1]
```diff
@@ -46,118 +46,118 @@
  * @see <a href="https://tools.ietf.org/html/rfc2295">RFC2295 (Transparent Content Negotiation)</a>
  * @since 4.0
  */
-public interface HttpStatus {
+public class HttpStatus {
```
Comment (michael-o) [inline httpcore5/src/main/java/org/apache/hc/core5/http/HttpStatus.java]: Also here maybe:

```
    private StandardCharsets() {
        throw new AssertionError("No java.nio.charset.StandardCharsets instances for you!");
    }
```

**Comment by michael-o** [comment]
@garydgregory Should this one be final?

**Comment by michael-o** [comment]
The branch is now uptodate and ready to be merged.

**Comment by michael-o** [comment]
Looks good. Please do a fixup and merge to master.

**Comment by michael-o** [comment]
@garydgregory Please do not merge unsquashed branches with a merge commit in the future. This produces a lot of noise and lowers the readability of the log.

**Comment by ok2c** [comment]
> @garydgregory Please do not merge unsquashed branches with a merge commit in the future. This produces a lot of noise and lowers the readability of the log.

Это какой то лютый п***ц

**Comment by michael-o** [comment]
> 
> 
> > @garydgregory Please do not merge unsquashed branches with a merge commit in the future. This produces a lot of noise and lowers the readability of the log.
> 
> Это какой то лютый п***ц

согласен!

**Pull request merged**